### ISSUE TITLE
MKL loaded in startup.jl makes init of LinearSolve fail on Julia 1.9.0-rc3 on Windows

### ISSUE THREAD

**Issue by jaakkor2** [SciML/LinearSolve.jl#300]
```
BLAS.get_config().loaded_libs[1].libname
```
fails on Julia 1.9.0-rc3 if MKL is loaded in `startup.jl` on Windows.  
```
ERROR: BoundsError: attempt to access 0-element Vector{LinearAlgebra.BLAS.LBTLibraryInfo} at index [1]
Stacktrace:
 [1] getindex(A::Vector{LinearAlgebra.BLAS.LBTLibraryInfo}, i1::Int64)
   @ Base .\essentials.jl:13
 [2] top-level scope
   @ REPL[2]:1
```
This makes `LinearSolve.jl` to fail on line https://github.com/SciML/LinearSolve.jl/blob/v1.42.0/src/init.jl#L6

Julia issue is here https://github.com/JuliaLang/julia/issues/49593 , but maybe LinearSolve could work around it?

**Comment by ChrisRackauckas** [comment]
@staticfloat how would we work around this?

**Comment by jaakkor2** [comment]
Fixed in Julia v1.9.1.
